skopt/parameter.py

```diff
@@ -50,59 +51,77 @@ def inverse_transform(self, values):
         return np.exp(values)
 
 
-class CategoryTransform(TransformerMixin):
+class Distribution:
+    def transform(self, values):
+        """Transform `values` from original into warped space."""
+        return self.transformer.transform(values)
+
+    def inverse_transform(self, values):
+        """Transform `values` from warped into original space."""
+        return self.transformer.inverse_transform(values)
+
+    @abc.abstractmethod
+    def rvs(self, n_samples=None, random_state=None):
+        """
+        Randomly sample points from the original space
+        """
+        return
+
+
+class CategoricalEncoder(TransformerMixin):
+    """
+    OneHotEncoder of scikit-learn that can handle categorical
+    variables.
+    """
     def __init__(self):
         """Convert labeled categories into one-hot encoded features"""
         self._label = LabelEncoder()
         self._onehot = OneHotEncoder()
 
     def fit(self, values):
+        """
+        Fit a list or array of categories.
+
+        Parameters
+        ----------
+        * `values` [array-like]:
+            List of categories.
+        """
         vals = np.asarray(values)
         vals = self._label.fit_transform(vals)
         self._onehot.fit(vals.reshape(-1, 1))
-
         return self
 
     def transform(self, values):
+        """
+        Transform an array of categories to a one-hot encoded representation.
+
+        Parameters
+        ----------
+        * `values` [array-like]:
+            List of categories.
+        """
         vals = np.asarray(values)
         vals = self._label.transform(vals).reshape(-1, 1)
         return self._onehot.transform(vals).toarray()
 
-    def inverse_transform(self, values):
-        vals = np.asarray(values)
-        vals = self._onehot.inverse_transform(vals)
-        return self._label.inverse_transform(vals)
-
-
-class Distribution:
-    def transform(self, values):
-        """Transform `values` from original into warped space."""
-        return self.transformer.transform(values)
-
-    def inverse_transform(self, values):
-        """Transform `values` from warped into original space."""
-        return self.transformer.inverse_transform(values)
-
-    def bounds(self):
-        return self.transform((self._low, self._high))
-
 
 class Real(Distribution):
     def __init__(self, low, high, prior='uniform', transformer='identity'):
         """Search space dimension that can take on any real value.
 
         Parameters
         ----------
-        low : float
+        * `low` [float]:
             Lower bound of the parameter.
 
-        high : float
+        * `high` [float]:
             Upper bound of the parameter.
 
-        prior : string or rv_frozen, default 'uniform'
+        * `prior` [string or rv_frozen, default='uniform']:
```

Comment:
Should we remove this argument for now? If the user provides say a Gaussian or Poisson prior, the low and high arguments become meaningless.